src/backend/joanie/core/models/contracts.py

```diff
@@ -281,3 +281,13 @@ def get_abilities(self, user):
         return {
             "sign": can_sign,
         }
+
+    def is_fully_signed(self):
+        """
+        Determine if a contract is fully signed by all parties. Call this method on the contract
+        instance. It returns a boolean indicating whether the contract is fully signed or not.
+        """
+        return self.submitted_for_signature_on is None and (
+            self.organization_signed_on is not None
+            and self.student_signed_on is not None
+        )
```

Comment:
```suggestion
        return self.organization_signed_on and self.student_signed_on and not self.submitted_for_signature_on
```
WDYT ?

Reply:
What you have suggested here was my first implementation and I kept on having bad surprises with `self.organization_signed_on` and `self.student_signed_on`. The only way I could make sure that no surprises occur, it was to set the condition explicitly and compare if the value is not `None`.

Maybe this would be clearer : 
```python
    @property
    def is_fully_signed(self):
        """
        Determine if a contract is fully signed by all parties. Call this method on the contract
        instance. It returns a boolean indicating whether the contract is fully signed or not.
        """
        return (
            self.organization_signed_on is not None
            and self.student_signed_on is not None
            and not self.submitted_for_signature_on
      )
```

Reply:
I prefer the first one.

Reply:
let's go for the 1st suggestion.